### ISSUE TITLE
Support customisable Instance names

### ISSUE THREAD

**Issue by mottosso** [pyblish/pyblish-qml#121]
Currently, an instance's name must be unique.

But this can result in poor display in the QML window. For example, you might have an instance in multiple families, each extracted into it's own format/representation. But because of this limitation, the display of instances must be unique, even though they are already visually separated via their family.

### Goal

Support customising these names, whilst still preserving the ability to reference an individual instance via the context.

This is already possible, since Pyblish 1.1.0.

```python
>>> import pyblish.api
>>> context = pyblish.api.Context()
>>> instance = context.create_instance("MyUniqueInstance")
>>> instance.data("name")  # Object name is returned, if no data exists
'MyUniqueInstance'
>>> instance.set_data("name", "My unique instance")
>>> instance.data("name")  # Now data exists
'My unique instance'
>>> context["MyUniqueInstance"]  # Initial name becomes an ID
Instance('MyUniqueInstance')
>>> context["My unique instance"]
KeyError
>>> instance.id
'MyUniqueInstance'
>>> instance.name
'MyUniqueInstance'
```

In QML, support this by showing the data name, when available. Else, revert to object name/id.

**Comment by tokejepsen** [comment]
Does this mean we can use the same named instances in different families?

**Comment by mottosso** [comment]
Exactly

**Comment by mottosso** [comment]
Implemented in [1.2](http://forums.pyblish.com/t/pyblish-1-2-released).
